kernel/arch/dreamcast/include/dc/maple.h

```diff
@@ -277,8 +276,10 @@ typedef struct maple_device {
     maple_frame_t           frame;          /**< \brief One rx/tx frame */
     struct maple_driver     *drv;           /**< \brief Driver which handles this device */
 
-    volatile int            status_valid;   /**< \brief Have we got our first status update? */
-    uint8                   status[1024];   /**< \brief Status buffer (for pollable devices) */
+    uint8                   probe_mask;     /**< \brief Mask of sub-devices left to probe */
+
+    volatile uint8          status_valid;   /**< \brief Have we got our first status update? */
+    uint8                   *status;        /**< \brief Status buffer (for pollable devices) */
```

Comment:
```suggestion
    uint8                   status[];        /**< \brief Status buffer (for pollable devices) */
```
Can we use this as just a FAM so we can over-allocate the entire struct as one contiguous segment in one-go instead of needing two `malloc()` calls?

EDIT: Shit, can't really gracefully change the allocation management to work with a FAM in the code review, so if you go with this method, you'll have to handle it yourself. lol.

Reply:
Resolved and force-pushed.